pages/multiplayer/[slug]/index.js

```diff
@@ -0,0 +1,364 @@
+import Pusher from "pusher-js";
+import axios from "axios";
+import { Chessboard } from "react-chessboard";
+import Chess from "chess.js";
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import Link from "next/link";
+import styled from "styled-components";
+import MoveInfoMultiplayer from "@/components//MoveInfoMultiplayer";
+import PlayerNameDisplay from "@/components/PlayerNameDisplay";
+import GameTerminal from "@/components/GameTerminal";
+import BoardWrapper from "@/components/BoardWrapper";
+
+//prevents undefined
+let pusher = null;
+
+export default function MultiPlayerPage({ username }) {
+  const [game, setGame] = useState(null);
+  const [moveStatus, setMoveStatus] = useState({});
+  const [fen, setFen] = useState(
+    "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1"
+  );
+  const [fenHistory, setFenHistory] = useState([]);
+  const [boardOrientation, setBoardOrientation] = useState("white");
+  const [showReplayBoard, setShowReplayBoard] = useState(false);
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  const [showChat, setShowChat] = useState(true);
+  const [messageToSend, setMessageToSend] = useState("");
+  const [chatStorage, setChatStorage] = useState([]);
+  const [onlineUsersCount, setOnlineUsersCount] = useState(0);
+  const [onlineUsers, setOnlineUsers] = useState([]);
```

Comment:
is looks like onlineUsersCount can be derived from onlineUsers and doesn't need to be its own state:

```suggestion
  const [onlineUsers, setOnlineUsers] = useState([]);
  
  const onlineUsersCount = onlineUsers.length
```